libshaderc/src/shaderc_test.cc

```diff
@@ -261,4 +261,74 @@ TEST(CompileString, MultipleThreadsCalling) {
   EXPECT_THAT(results, Each(true));
 }
 
+TEST(CompileKinds, Vertex) {
+  Compiler compiler;
+  ASSERT_NE(nullptr, compiler.get_compiler_handle());
+  const std::string kVertexShader = "void main(){ gl_Position = vec4(0);}";
+  EXPECT_TRUE(CompilationSuccess(compiler.get_compiler_handle(), kVertexShader,
+                                 shaderc_glsl_vertex_shader));
+}
+
+TEST(CompileKinds, Fragment) {
+  Compiler compiler;
+  ASSERT_NE(nullptr, compiler.get_compiler_handle());
+  const std::string kFragShader = "void main(){ gl_FragColor = vec4(0);}";
+  EXPECT_TRUE(CompilationSuccess(compiler.get_compiler_handle(), kFragShader,
+                                 shaderc_glsl_fragment_shader));
+}
+
+TEST(CompileKinds, Compute) {
+  Compiler compiler;
+  ASSERT_NE(nullptr, compiler.get_compiler_handle());
+  const std::string kCompShader =
+      "#version 310 es\n"
+      "void main(){}\n";
+  EXPECT_TRUE(CompilationSuccess(compiler.get_compiler_handle(), kCompShader,
+                                 shaderc_glsl_compute_shader));
+}
+
+TEST(CompileKinds, Geometry) {
+  Compiler compiler;
+  ASSERT_NE(nullptr, compiler.get_compiler_handle());
+  const std::string kGeoShader =
```

Comment:
That's a new syntax for me, but sure. Note that due to the GLES requirement of no characters prior to the #version token, the R"( had to go on the same line as the first shader line.